src/App.js

```diff
@@ -28,31 +28,92 @@ const generateSquares = () => {
 const App = () => {
 
   const [squares, setSquares] = useState(generateSquares());
+  const [userTurn, setTurn] = useState(PLAYER_1);
+  const [winner, setWinner] = useState("In progress!");
 
-  // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  const winCombo = [
+    [0,1,2],[3,4,5],[6,7,8],
+    [0,3,6],[1,4,7],[2,5,8],
+    [0,4,8],[2,4,6]
+  ]
 
+  const onClickCallback = (event) => {
+    if (winner === "In progress!") {
+      const newSquares = squares.map(row =>
+        row.map(square =>
+          square.id == [event.target.id] && square.value === '' ? 
```

Comment:
It looks like you used a `==` by accident here.  Remember to pay attention to the warnings that you get in the terminal.

```
./src/App.js
  Line 44:21:  Expected '===' and instead saw '=='  eqeqeq
```

```suggestion
          square.id === [event.target.id] && square.value === '' ? 
```